.github/workflows/default.yaml

```
@@ -94,14 +94,26 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04, macos-14 ]
+        os:
+          - ubuntu-22.04
+          - macos-14
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
         layer:
-          - layer-00-default
-          - layer-01-minimal
-          - layer-02-maximus
-          - layer-04-noauth-everything
+          - { layer: layer-00-default, nick: default }
+          - { layer: layer-01-minimal, nick: minimal }
+          - { layer: layer-02-maximus, nick: maximus }
+          - { layer: layer-04-noauth-everything, nick: noauth }
```

Please avoid visually odd layer.layer names. I can suggest using "name" for the inner "layer" field:

```suggestion
          - { name: layer-00-default, nick: default }
          - { name: layer-01-minimal, nick: minimal }
          - { name: layer-02-maximus, nick: maximus }
          - { name: layer-04-noauth-everything, nick: noauth }
```

Another reasonable option is "test-builds-parameter" (because that is why we need layer.layer value; it is how it is used).